prov/efa/src/efa_hmem.c

```diff
@@ -276,70 +227,81 @@ static int efa_domain_hmem_info_init_neuron(struct efa_domain *efa_domain)
 	info->p2p_supported_by_device = true;
-	efa_domain_hmem_info_init_protocol_thresholds(efa_domain, FI_HMEM_NEURON);
-	if (-FI_ENODATA != fi_param_get(&efa_prov, "inter_max_medium_message_size", &tmp_value)) {
-		EFA_WARN(FI_LOG_DOMAIN,
-		         "The environment variable FI_EFA_INTER_MAX_MEDIUM_MESSAGE_SIZE was set, "
-		         "but EFA HMEM via Neuron API only supports eager and runting read protocols. "
-				 "The variable will not modify Neuron memory run config.\n");
-	}
-
+	return;
 #endif
-	return 0;
+	return;
+}
+
+static inline bool efa_domain_hmem_info_is_p2p_supported_by_device(struct efa_domain *efa_domain, enum fi_hmem_iface iface) {
+	struct efa_hmem_info *info = &efa_domain->hmem_info[iface];
+	switch (iface) {
+	case FI_HMEM_CUDA:
+		efa_domain_hmem_info_is_p2p_supported_by_cuda(info);
+		return info->p2p_supported_by_device;
+	case FI_HMEM_NEURON:
+		efa_domain_hmem_info_is_p2p_supported_by_neuron(info);
+		return info->p2p_supported_by_device;
+	case FI_HMEM_SYNAPSEAI:
+		return true;
+	default:
+		return false;
+	}
 }
 
 /**
- * @brief          Initialize the efa_hmem_info state for FI_HMEM_SYNAPSEAI
+ * @brief Initialize the efa_hmem_info state for iface
  *
  * @param[in,out]  efa_domain  Pointer to struct efa_domain
- *
- * @return         0
+ * @param[in]      iface       HMEM interface
  */
-static int efa_domain_hmem_info_init_synapseai(struct efa_domain *efa_domain)
+static void
+efa_domain_hmem_info_init_iface(struct efa_domain *efa_domain, enum fi_hmem_iface iface)
 {
-#if HAVE_SYNAPSEAI
-	size_t tmp_value;
-	struct efa_hmem_info *info = &efa_domain->hmem_info[FI_HMEM_SYNAPSEAI];
+	struct efa_hmem_info *info = &efa_domain->hmem_info[iface];
 
-	if (!ofi_hmem_is_initialized(FI_HMEM_SYNAPSEAI)) {
-		EFA_INFO(FI_LOG_DOMAIN, "FI_HMEM_SYNAPSEAI is not initialized\n");
-		return 0;
+	if (!ofi_hmem_is_initialized(iface)) {
+		EFA_INFO(FI_LOG_DOMAIN, "%s is not initialized\n",
+		         fi_tostr(&iface, FI_TYPE_HMEM_IFACE));
+		return;
 	}
 
-	if (!(g_device_list[0].device_caps & EFADV_DEVICE_ATTR_CAPS_RDMA_READ)) {
+	if ((iface == FI_HMEM_SYNAPSEAI || iface == FI_HMEM_NEURON) &&
+	    !(g_device_list[0].device_caps & EFADV_DEVICE_ATTR_CAPS_RDMA_READ)) {
 		EFA_WARN(FI_LOG_DOMAIN,
-			 "No EFA RDMA read support, transfers using Habana Gaudi will fail.\n");
-		return 0;
+			 "No EFA RDMA read support, transfers using %s will fail.\n",
+			 fi_tostr(&iface, FI_TYPE_HMEM_IFACE));
+		return;
 	}
 
 	info->initialized = true;
-	info->p2p_disabled_by_user = false;
-	/* SynapseAI currently requires P2P */
-	info->p2p_required_by_impl = true;
-	info->p2p_supported_by_device = true;
-	efa_domain_hmem_info_init_protocol_thresholds(efa_domain, FI_HMEM_SYNAPSEAI);
+	info->p2p_disabled_by_user = ofi_hmem_p2p_disabled();
+	info->p2p_supported_by_device = info->p2p_disabled_by_user ?
+		false : efa_domain_hmem_info_is_p2p_supported_by_device(efa_domain, iface);
+	if (iface == FI_HMEM_SYNAPSEAI)
+		info->p2p_supported_by_device = true;
```

Comment:
Should remove.

Reply:
SYNAPSEAI always support p2p, so this should be true irrelevant of p2p disabled status.

Reply:
This is not necessary. If the user disables p2p on synapseai platform, then we should not attempt to use synapseai p2p, so it does not matter whether the device supports it or not.

This is the same for CUDA/Neuron - even if the device supports p2p, we still set p2p_supported_by_device = false it the user disables it.

Reply:
`p2p_supported_by_device` still needs to be set for the p2p check later right? If we remove it here, do we have it somewhere else?

Reply:
We set it in the function efa_domain_hmem_info_is_p2p_supported_by_device
```
info->p2p_supported_by_device = !info->p2p_disabled_by_user && efa_domain_hmem_info_is_p2p_supported_by_device(efa_domain, iface);
```